packages/arb-node-core/nodehealth/nodehealth.go

```diff
@@ -0,0 +1,269 @@
+	Config bool
+
+	ValStr    string
+	ValBigInt big.Int
+	ValTime   time.Duration
+}
+
+type inboxReaderState struct {
+	getNextBlockToRead big.Int
+	currentHeight      big.Int
+	arbCorePosition    big.Int
+	caughtUpTarget     big.Int
+}
+
+type healthState struct {
+	mu          sync.Mutex
+	inboxReader inboxReaderState
+}
+
+type aSyncDataStruct struct {
+	//Healthchecks to allow the status to be shared between handlers
+	checkOpenethereum healthcheck.Check
+	checkPrimary      healthcheck.Check
+	inboxReaderStatus healthcheck.Check
+}
+
+func updateInboxReader(state *healthState, logMessage Log) {
+	//Load the log into the correct struct field inside the state array
+	if logMessage.Var == "getNextBlockToRead" {
+		state.mu.Lock()
+		state.inboxReader.getNextBlockToRead = logMessage.ValBigInt
+		state.mu.Unlock()
+	}
+	if logMessage.Var == "currentHeight" {
+		state.mu.Lock()
+		state.inboxReader.currentHeight = logMessage.ValBigInt
+		state.mu.Unlock()
+	}
+	if logMessage.Var == "arbCorePosition" {
+		state.mu.Lock()
+		state.inboxReader.arbCorePosition = logMessage.ValBigInt
+		state.mu.Unlock()
+	}
+	if logMessage.Var == "caughtUpTarget" {
+		state.mu.Lock()
+		state.inboxReader.caughtUpTarget = logMessage.ValBigInt
+		state.mu.Unlock()
+	}
+}
+
+func updateConfig(config *configStruct, logMessage Log) {
+	//Load the configuration message into the config struct
+	if logMessage.Var == "openethereumHealthcheckRPC" {
+		config.mu.Lock()
+		config.openethereumHealthcheckRPC = logMessage.ValStr
+		config.mu.Unlock()
+	}
+	if logMessage.Var == "primaryHealthcheckRPC" {
+		config.mu.Lock()
+		config.primaryHealthcheckRPC = logMessage.ValStr
+		config.mu.Unlock()
+	}
+}
+
+func logger(config *configStruct, state *healthState, logMsgChan <-chan Log) {
+	for {
+		//Read log structure from channel
+		logMessage := <-logMsgChan
+		//Check if a configuration message has been sent
+		if logMessage.Config {
+			updateConfig(config, logMessage)
+		} else {
+			//Check if the InboxReader is sending logs
+			if logMessage.Comp == "InboxReader" {
+				updateInboxReader(state, logMessage)
+			}
+		}
+	}
+}
+
+// Default configuration values for the healthcheck server
+func (config *configStruct) loadConfig() {
+	const defaultSuccessCode = 200
+	const defaultBlockDifferenceTolerance = 2
+	const defaultPollingRate = 10 * time.Second
+	const loopDelayTimer = 1 * time.Second
+	config.healthcheckRPC = "0.0.0.0:8080"
+	config.pollingRate = defaultPollingRate
+	config.loopDelayTimer = loopDelayTimer
+	config.openethereumHealthcheckRPC = ""
+	config.primaryHealthcheckRPC = ""
+	config.successCode = defaultSuccessCode
+	config.blockDifferenceTolerance = defaultBlockDifferenceTolerance
+}
+
+func checkEndpoint(config *configStruct, endpoint string) healthcheck.Check {
+	check := healthcheck.Async(func() error {
+		//Lock config mutex for read operation
+		config.mu.Lock()
+		//Retrieve status code from healthcheck endpoint
+		res, err := http.Get(endpoint + "/ready")
```

Comment:
I changed endpoint to being passed through a pointer.  The lock should be required since the channel listener could write to it during a read which would cause an error I thought?